app/utils/nonsipp/summary/LoansCheckAnswersUtils.scala

```diff
@@ -730,20 +802,58 @@ trait LoansCheckAnswersUtils(schemeDateService: SchemeDateService)
   private def loanOutstandingSection(
     srn: Srn,
     index: Max5000,
-    arrearsPrevYears: Option[Boolean],
-    outstandingArrearsOnLoan: Option[Money],
+    arrearsPrevYears: Either[String, Boolean],
+    outstandingArrearsOnLoan: Either[String, Option[Money]],
     returnEndDate: LocalDate,
     mode: Mode
   ): List[CheckYourAnswersSection] = {
+
+    val isDataIncomplete =
+      arrearsPrevYears.isLeft ||
+        (arrearsPrevYears == Right(true) && (outstandingArrearsOnLoan.isLeft || outstandingArrearsOnLoan.exists(
+          _.isEmpty
+        )))
+
+    val header = if (isDataIncomplete) {
+      Some(Heading2.medium("loansCheckAndUpdate.isPrePop.section6.heading"))
+    } else {
+      Some(Heading2.medium("loanCheckYourAnswers.section6.heading"))
+    }
+
     val outstandingMessage = arrearsPrevYears match {
-      case Some(true) => "site.yes"
-      case Some(false) => "site.no"
-      case None => ""
+      case Right(true) => "site.yes"
+      case Right(false) => "site.no"
+      case Left(value) => s"$value"
     }
 
+    val maybeOutstandingArrearsRow: Option[CheckYourAnswersRowViewModel] =
+      (arrearsPrevYears, outstandingArrearsOnLoan) match {
+        case (Right(true), Right(maybeMoney)) =>
+          val valueText = maybeMoney match {
+            case Some(value) => s"£${value.displayAs}"
+            case None => "Incomplete"
+          }
+
+          Some(
+            CheckYourAnswersRowViewModel(
+              "loanCheckYourAnswers.section6.arrears.yes",
+              valueText
+            ).withAction(
+              SummaryAction(
+                "site.change",
+                controllers.nonsipp.loansmadeoroutstanding.routes.OutstandingArrearsOnLoanController
+                  .onPageLoad(srn, index, mode)
+                  .url + "#details"
+              ).withVisuallyHiddenContent("loanCheckYourAnswers.section6.arrears.yes.hidden")
+            )
+          )
+
+        case _ => None
+      }
+
     List(
       CheckYourAnswersSection(
-        Some(Heading2.medium("loanCheckYourAnswers.section6.heading")),
+        header,
```

Comment:
There are quite a few conditional displays introduced here.
@narendrapaduchuri could you please discuss the changes with @ibadmus and/or @rameshmandanapu1 to make sure that nothing is broken for these sections that were changed?